### PR TITLE
feat(filtertargs): add prop for overflow menu text

### LINKED ISSUE #3710: [FilterTags] change text for overflow items tag
### What package is this for?

- [x] React

### Summary

Replace text for overflow items in `FilterTags`

**Additional context**

Currently, overflowed tags will be displayed as tag with  text: `More: 3` (see screenshot).
New implementation should display simple `+3`.

Screenshot 2023-02-03 at 19 21 19

### PR DESCRIPTION
**Pull request by erzhan-temir-mamyrov**
Closes #3710

**Summary**

- Add i18n prop for text in OverflowMenu displayed for truncated FilterTags

**Change List (commits, features, bugs, etc)**

- Add i18n prop to FilterTags component
- Add default i18n prop for FilterTags inside Table component
- Update tests and docs

**Acceptance Test (how to verify the PR)**

- Go to [this story](https://deploy-preview-3713--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-filtering)
- Enable: `hasAdvancedFilters`
- Click on `Enable overflow menu in filter tags for advanced filters`
- Re-enable `hasAdvancedFilters` in order to reset memoized filters
- See OverflowMenu for truncated filters displayed as `+1` (resize screen horizontally if it's not displayed)

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/FilterTags/FilterTags.jsx [hunk 1]
```diff
@@ -78,6 +90,11 @@ const FilterTags = ({ children, hasOverflow, id, tagContainer, onChange, i18n, t
     /* eslint-enable react-hooks/exhaustive-deps */
   );
 
+  const filterTagsOverflowMenuTextText = filterTagsOverflowMenuText.replace(
```
Comment (herleraja) [inline packages/react/src/components/FilterTags/FilterTags.jsx:93]: image
better to use it as a callback function. so if they need any format they can use.

Comment (erzhan-temir-mamyrov) [reply packages/react/src/components/FilterTags/FilterTags.jsx]: @herleraja Thanks for the tip! I replaced string with the callback function.

**Comment by davidicus** [comment]
There is some issues with the overflow of the text. 

image

File: packages/react/src/components/Table/TablePropTypes.js [hunk 2]
```diff
@@ -225,6 +225,8 @@ export const I18NPropTypes = PropTypes.shape({
   toolbarTooltipLabel: PropTypes.string,
   /** tooltip text for filter row icon button */
   filterRowIconDescription: PropTypes.string,
+  /** overflow menu text callback for truncated filter tags */
+  filterTagsOverflowMenuText: PropTypes.func,
```
Comment (davidicus) [inline packages/react/src/components/Table/TablePropTypes.js:229]: We should open this up to both string and function. Graphite does not use function translations.

**Review by davidicus — CHANGES_REQUESTED** [review]
Need to open up the prop-types.

File: packages/react/src/components/Table/TablePropTypes.js (hunk 2 above)
Comment (erzhan-temir-mamyrov) [reply packages/react/src/components/Table/TablePropTypes.js]: @davidicus Thanks for the heads up! I added string for translation as well. Also, I fixed text overflow issue.

**Review by herleraja — APPROVED** [review]
LGTM

**Review by davidicus — APPROVED** [review]

**Review by davidicus — APPROVED** [review]

**Pull request merged**